rm-config/src/keymap.rs

```diff
@@ -0,0 +1,425 @@
+    fn to_str(self) -> &'static str {
+        match self {
+            KeyModifier::None => "",
+            KeyModifier::Ctrl => "CTRL",
+            KeyModifier::Shift => "SHIFT",
+            KeyModifier::Alt => "ALT",
+            KeyModifier::Super => "SUPER",
+            KeyModifier::Meta => "META",
+        }
+    }
+
+    fn is_none(self) -> bool {
+        self == KeyModifier::None
+    }
+}
+
+impl From<KeyModifier> for KeyModifiers {
+    fn from(value: KeyModifier) -> Self {
+        match value {
+            KeyModifier::None => KeyModifiers::NONE,
+            KeyModifier::Ctrl => KeyModifiers::CONTROL,
+            KeyModifier::Shift => KeyModifiers::SHIFT,
+            KeyModifier::Alt => KeyModifiers::ALT,
+            KeyModifier::Super => KeyModifiers::SUPER,
+            KeyModifier::Meta => KeyModifiers::META,
+        }
+    }
+}
+
+impl Default for KeyModifier {
+    fn default() -> Self {
+        Self::None
+    }
+}
+
+impl KeymapConfig {
+    pub const FILENAME: &'static str = "keymap.toml";
+    const DEFAULT_CONFIG: &'static str = include_str!("../defaults/keymap.toml");
+
+    pub fn init() -> Result<Self> {
+        match utils::fetch_config::<Self>(Self::FILENAME) {
+            Ok(mut keymap_config) => {
+                keymap_config.populate_hashmap();
+                return Ok(keymap_config);
+            }
+            Err(e) => match e {
+                utils::ConfigFetchingError::Io(e) if e.kind() == ErrorKind::NotFound => {
+                    let mut keymap_config =
+                        utils::put_config::<Self>(Self::DEFAULT_CONFIG, Self::FILENAME)?;
+                    keymap_config.populate_hashmap();
+                    return Ok(keymap_config);
+                }
+                utils::ConfigFetchingError::Toml(_) => anyhow::bail!(e),
+                _ => anyhow::bail!(e),
+            },
+        }
+    }
+
+    pub fn get_keys_for_action(&self, action: Action) -> Option<String> {
+        let mut keys = vec![];
+
+        for keybinding in &self.general.keybindings {
+            if action == keybinding.action.into() {
+                keys.push(keybinding.keycode_string());
+            }
+        }
+        for keybinding in &self.torrents_tab.keybindings {
+            if action == keybinding.action.into() {
+                keys.push(keybinding.keycode_string());
+            }
+        }
+
+        if keys.is_empty() {
+            return None;
+        } else {
+            Some(keys.join("/"))
+        }
+    }
+
+    fn populate_hashmap(&mut self) {
+        for keybinding in &self.general.keybindings {
+            let hash_value = (keybinding.on, keybinding.modifier.into());
+            self.keymap.insert(hash_value, keybinding.action.into());
+        }
+        for keybinding in &self.torrents_tab.keybindings {
+            let hash_value = (keybinding.on, keybinding.modifier.into());
+            self.keymap.insert(hash_value, keybinding.action.into());
+        }
```

Comment:
possible to create an iterator of `Iter<Keybinding>` and iterate through a concatenation of both? in case we add more keybindings in the future

Reply:
Great idea

Reply:
Actually, there's a better way to do this, and it would get rid of generics in the code, but that would require do refactorings all over rm-main so that'll be one of my next PRs.